File: frontend/src/app/models/fasten/resource_association.ts

export class ResourceAssociation {
  source_id: string
  source_resource_type: string
  source_resource_id: string

  old_related_source_id?: string
  old_related_source_resource_type?: string
  old_related_source_resource_id?: string


  new_related_source_id: string
  new_related_source_resource_type: string
  new_related_source_resource_id: string
}

File: backend/pkg/models/resource_association.go
package models

type ResourceAssociation struct {
	SourceID           string `json:"source_id"`
	SourceResourceType string `json:"source_resource_type"`
	SourceResourceID   string `json:"source_resource_id"`

	OldRelatedSourceID           string `json:"old_related_source_id"`
	OldRelatedSourceResourceType string `json:"old_related_source_resource_type"`
	OldRelatedSourceResourceID   string `json:"old_related_source_resource_id"`

	NewRelatedSourceID           string `json:"new_related_source_id"`
	NewRelatedSourceResourceType string `json:"new_related_source_resource_type"`
	NewRelatedSourceResourceID   string `json:"new_related_source_resource_id"`
}
